walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,140 +14,180 @@
 
 package com.google.sps;
 
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.LinkedList;
+import java.util.ListIterator;
 
 public final class FindMeetingQuery {
+  private int meetingDuration;
 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
 
     // The total (common) free time we have to conduct the meeting.
-    ArrayList<TimeRange> free_time = new ArrayList<TimeRange>();
-    free_time.addAll(Arrays.asList(TimeRange.WHOLE_DAY));
+    LinkedList<TimeRange> freeTime = new LinkedList<TimeRange>();
+    freeTime.addAll(Arrays.asList(TimeRange.WHOLE_DAY));
 
-    int meeting_duration = (int) request.getDuration();
+    meetingDuration = (int) request.getDuration();
+    LinkedList<String> optionalAttendees = new LinkedList<>(request.getOptionalAttendees());
 
     // If duration of the meeting is greater than a whole day, then return an empty list.
-    if (meeting_duration > TimeRange.WHOLE_DAY.duration()) {
+    if (meetingDuration > TimeRange.WHOLE_DAY.duration()) {
       return Arrays.asList();
     }
 
     // If there are no attendees then return the whole day (ie the free time we have till now).
-    if (request.getAttendees().isEmpty()) {
-      return free_time;
+    if (request.getAttendees().isEmpty() && optionalAttendees.isEmpty()) {
+      return freeTime;
     }
 
-    /**
-     * Go through every event. If an event has attendees of the requested meeting, then check if the
-     * time overlaps with our free time. If yes then remove that conflicted time slot from our free
-     * time.
+    // To store all the events which the optional attendees are attending.
+    LinkedList<Event> optionalAttendeeEvents = new LinkedList<>();
+
+    /** Changes freeTime such that it now conatins all the free time slots when the meeting can
```

Comment:
Starting your comment with `/**` is only used for JavaDoc and should not be used inside of a function. Please use `/*` unless you write JavaDoc for your functions.

Reply:
Ohh okay, I think I'm supposed to know that.
Thanks for letting me know.
I made a note of it and will remember to follow that from now on